src/endpoints/sources.js

```diff
@@ -202,6 +202,39 @@ const putDataset = contentTypesConsumed([
 ]);
 
 
+/* DELETE
+ */
+
+/**
+ * Generate an Express endpoint that deletes a dataset or narrative Resource
+ * determined by the request.
+ *
+ * Internally, the Resource is deleted by deleting all of its Subresources.
+ *
+ * @param {resourceExtractor} resourceExtractor - Function to provide the Resource instance from the request
+ * @returns {expressEndpointAsync}
+ */
+const deleteResource = resourceExtractor => async (req, res) => {
+  const resource = resourceExtractor(req);
+
+  const method = "DELETE";
+  const upstreamUrls = await Promise.all(resource.subresources().map(s => s.url(method)));
+  const upstreamResponses = await Promise.all(upstreamUrls.map(url => fetch(url, {method})));
+
+  const goodStatuses = new Set([204, 404]);
+
+  if (!upstreamResponses.every(r => goodStatuses.has(r.status))) {
+    throw new InternalServerError("failed to delete all subresources");
```

Comment:
>  I don't think a user cares about which subresources get deleted or not. A user just wants the dataset to be gone.

Ah, that's true. I was still in the S3 mindset of managing individual files rather than a whole dataset. 
Good to know we have logs to surface these errors!